File: DSA/Java/test/dsa/lib/algo/sort/MergeSortTest.java
package dsa.lib.algo.sort;

import org.junit.Test;

import static org.junit.Assert.*;

public class MergeSortTest {
    private static void mergeSort(int[] ary) {
        new MergeSort().sort(ary, ary.length);
    }

    @Test
    public void oneNum() {
        int[] ary = {1};
        final int[] expected = {1};
        mergeSort(ary);
        assertArrayEquals(expected, ary);
    }

    @Test
    public void case1() {
        int[] ary = {4, 5, 6, 3, 2, 1};
        final int[] expected = {1, 2, 3, 4, 5, 6};
        mergeSort(ary);
        assertArrayEquals(expected, ary);
    }

    @Test
    public void case2() {
        int[] ary = {12, 11, 13, 5, 6, 7};
        final int[] expected = {5, 6, 7, 11, 12, 13};
        mergeSort(ary);
        assertArrayEquals(expected, ary);
    }
}

File: DSA/Cpp/test/dsa/lib/algo/sort/merge_sort_test.cpp
#include <dsa/lib/algo/sort/merge_sort.hpp>
#include "sort_test.h"
namespace dsa::lib::algo::sort {
namespace {

class MergeSortTest : public SortTest {
 public:
    static void sort(int *ary, length_t n) {
        MergeSort ms;
        ms.sort(ary, n);
    }
};
TEST_F(MergeSortTest, case1) {
    int ary[] = {1};
    int expected[] = {1};
    sort(ary, arylen(ary));
    EXPECT_ARRAY_EQ(expected, arylen(expected), ary, arylen(ary));
}
TEST_F(MergeSortTest, case2) {
    int ary[] = {4, 5, 6, 3, 2, 1};
    int expected[] = {1, 2, 3, 4, 5, 6};
    sort(ary, arylen(ary));
    EXPECT_ARRAY_EQ(expected, arylen(expected), ary, arylen(ary));
}
TEST_F(MergeSortTest, case3) {
    int ary[] = {12, 11, 13, 5, 6, 7};
    int expected[] = {5, 6, 7, 11, 12, 13};
    sort(ary, arylen(ary));
    EXPECT_ARRAY_EQ(expected, arylen(expected), ary, arylen(ary));
}

}
}
